azure-iot-device/azure/iot/device/common/mqtt_transport.py

```diff
@@ -420,7 +420,9 @@ def reauthorize_connection(self, password=None):
         try:
             rc = self._mqtt_client.reconnect()
         except Exception as e:
-            raise exceptions.ProtocolClientError(
+            logger.info("reconnect raised {}".format(e))
+            self._stop_automatic_reconnect()
```

Comment:
>_stop_automatic_reconnect [](start = 17, length = 25)

Note to self: need to call _stop_automatic_reconnect on publish/subscribe/unsubscribe failures also

Reply:
Is this coming in a future PR then? Why not now?

---
In reply to: [387699653](https://github.com/Azure/azure-iot-sdk-python/pull/476#discussion_r387699653) [](ancestors = 387699653)

Reply:
Also why did this become a ConnectionDroppedError instead of a ProtocolClientError? I don't quite follow

---
In reply to: [387968511](https://github.com/Azure/azure-iot-sdk-python/pull/476#discussion_r387968511) [](ancestors = 387968511,387699653)

Reply:
Why not now?  because I'm not 100% sure this is even a real problem or just theoretical.  I don't want to delay a PR that solves real problems with the possibility that there might be other vaguely related problems.

Why ConnectionDroppedError?  It became ConnectionDroppedError because that matches the behavior we see.  If reconnect() raises, it's most likely because the connection was dropped.  "ProtocolClientError" is a catch-all that means "something went wrong, but we don't know what"

I'm going to go back and see if I can narrow the set of cases where we raise ConnectionDroppedError.  The problem is that we have no real recourse on ProtocolClientError, but we do have a recovery path for ConnectionDroppedError.

Our longer-term goal should be to get rid of ProtocolClientError entirely, or at least reserve it for cases where we know how to remedy the situation (perhaps by rebooting).  Right now, our ProtocolClientError handling goes along the lines of  "something went wrong, but we don't know what, so we're going to return a failure and hope everything is fine."  If we can actually fix the situation that lead to the error (by reconnecting, for instance), we should.  



---
In reply to: [388035883](https://github.com/Azure/azure-iot-sdk-python/pull/476#discussion_r388035883) [](ancestors = 388035883,387968511,387699653)